### PR TITLE
Improve efficiency of the test suite

### PR DESCRIPTION
**Pull request by flferretti**
This pull request includes several updates to the test suite to improve the code style and likely bringing minor improvements in terms of performance.
These changes will enhance the overall efficiency and maintainability of the codebase.

----
📚 Documentation preview 📚: https://jaxsim--166.org.readthedocs.build//166/

### CODE REVIEW

File: tests/utils_idyntree.py [hunk 1]
```diff
@@ -127,9 +126,7 @@ def build(
         urdf: pathlib.Path | str,
         considered_joints: list[str] = None,
         vel_repr: VelRepr = VelRepr.Inertial,
-        gravity: npt.NDArray = dataclasses.field(
-            default_factory=lambda: np.array([0, 0, -10.0])
-        ),
+        gravity: npt.NDArray = np.array([0, 0, -10.0]),
```
Comment (diegoferigo) [inline tests/utils_idyntree.py:129]: I'm only wondering why this works. Recent Python versions no longer allow this kind of initialization with mutable objects. If I recall, also numpy arrays were affected.

Comment (flferretti) [reply tests/utils_idyntree.py:129]: I guess it's because this is the argument of a method to build the class, not the attribute declaration

Comment (diegoferigo) [reply tests/utils_idyntree.py:129]: Right :) I Didn't notice I was commenting on the `build` method :) I thought it was the attribute initialization, my bad.

**Review by diegoferigo — APPROVED** [review]

File: tests/utils_idyntree.py (hunk 1 above)
Comment (diegoferigo) [reply tests/utils_idyntree.py:129]: I guess the original code was a copy-paste leftover, woops.

**Pull request merged**